### ISSUE TITLE
Improve naming/structure of configs options

### ISSUE THREAD

**Issue by NicolasLiampotis** [rciam/rciam-federation-registry-agent#15]
We need better names for the config options. We can look into two different approaches:

### 1 Keep flat structure but use naming convention
- [ ] ARGO Messaging (AMS) options will be prefixed with `ams_`
- [ ] SimpleSAMLphp (SSP) options will be prefixed with `ssp_`
- [x] MITREid Connect options will be prefixed with `mitreid_`

#### Agent-specific changes

- `metadata` --> `ssp_metadata_conf_file`
- `metadata_key` --> `ssp_cron_secret`
- `ssp_url` --> `ssp_cron_url`
-  MISSING --> `ssp_cron_tag`
- `mitreid_url` --> `mitreid_issuer`

### 2 Move to structured config

```
{
  "ams": { // ARGO Messaging options
    "host": "example.host.com",
    "project" : "ams-project-name",
    "pull_topic" : "ams-topic",
    "pull_sub" : "ams-sub",
    "token" : "ams-token",
    "pub_topic": "ams-publish-topic",
    "poll_interval": 1 // AMS poll interval in seconds
  },
  "ssp": { // SimpleSAMLphp options
    "metadata_conf_file": "/path/to/ssp/metadata/file.php",
    "cron_secret": "SSP cron secret",
    "cron_url": "http://localhost/proxy/module.php/cron/cron.php",
    "cron_tag": "hourly"
  },
  "mitreid": { // MITREid Connect options
    "issuer": "https://example.com/oidc" ,
    "refresh_token": "refresh token",
    "client_id": "client ID",
    "client_secret": "client secret"
  },
  "log_conf": "conf/logger.conf"
}
```

/cc @kevangel79 @Tas-sos @linathedog

**Comment by kevangel79** [comment]
I think the option#2 is much cleaner and distincts each config item better.
However, we still have to use 2 different configuration files because of the different ams pull topics and subs that the 2 deployers use.

So, the configs will look like the following examples. 

**ssp_deployer_config.json**
```
{
  "ams": {
    "host": "example.host.com",
    "project" : "ams-project-name",
    "pull_topic" : "ams-topic",
    "pull_sub" : "ams-sub",
    "token" : "ams-token",
    "pub_topic": "ams-publish-topic",
    "poll_interval": 1
  },
  "ssp": {
    "metadata_conf_file": "/path/to/ssp/metadata/file.php",
    "cron_secret": "SSP cron secret",
    "cron_url": "http://localhost/proxy/module.php/cron/cron.php",
    "cron_tag": "hourly"
  },
  "log_conf": "conf/logger.conf"
}
```

**mitreid_deployer_config.json**
```
{
  "ams": {
    "host": "example.host.com",
    "project" : "ams-project-name",
    "pull_topic" : "ams-topic",
    "pull_sub" : "ams-sub",
    "token" : "ams-token",
    "pub_topic": "ams-publish-topic",
    "poll_interval": 1
  },
  "mitreid": {
    "issuer": "https://example.com/oidc" ,
    "refresh_token": "refresh token",
    "client_id": "client ID",
    "client_secret": "client secret"
  },
  "log_conf": "conf/logger.conf"
}
```

**Comment by NicolasLiampotis** [comment]
I'm also in favour of option 2. There is a 3rd option to avoid separate config files:

```
{
  "ssp": {
    "ams": {
      "host": "example.host.com",
      "project" : "ams-project-name-ssp",
      "pull_topic" : "ams-topic-ssp",
      "pull_sub" : "ams-sub-ssp",
      "token" : "ams-token-ssp",
      "pub_topic": "ams-publish-topic-ssp",
      "poll_interval": 1
    },
    "metadata_conf_file": "/path/to/ssp/metadata/file.php",
    "cron_secret": "SSP cron secret",
    "cron_url": "http://localhost/proxy/module.php/cron/cron.php",
    "cron_tag": "hourly"
    //"log_conf": "conf/logger.conf" UNCOMMENT if we need per agent log conf
  },
  "mitreid": {
    "ams": {
      "host": "example.host.com",
      "project" : "ams-project-name-mitreid",
      "pull_topic" : "ams-topic-mitreid",
      "pull_sub" : "ams-sub-mitreid",
      "token" : "ams-token-mitreid",
      "pub_topic": "ams-publish-topic-mitreid",
      "poll_interval": 1
    },
    "issuer": "https://example.com/oidc" ,
    "refresh_token": "refresh token",
    "client_id": "client ID",
    "client_secret": "client secret"
    //"log_conf": "conf/logger.conf" UNCOMMENT if we need per agent log conf
  },
  "log_conf": "conf/logger.conf" // GLOBAL log conf
}
```
Thoughts?

**Comment by kevangel79** [comment]
Should we implement a generic fallback to global functionality for every item or just for `log_conf` ?

**Comment by NicolasLiampotis** [comment]
To keep things simple I would prefer a single global config for logging. The only real world scenario I can think of is enabling debug for a specific agent. What else would we lose?

**Comment by kevangel79** [comment]
I agree, this will keep it simple and with option#3 we have only a single conf to maintain. Should i proceed with the changes ?

**Comment by Tas-sos** [comment]
Very nice and very interesting proposal!
* I agree with the addition of the **prefix** - it will be clearer and more human readable. 
* I also agree with [option#3](https://github.com/rciam/rciam-federation-registry-agent/issues/15#issuecomment-750295782) with only one **clean** configuration file.

**Comment by NicolasLiampotis** [comment]
Resolved with PR #20
